flink-connectors/flink-connector-mongodb/src/main/java/org/apache/flink/mongodb/table/source/MongodbDynamicTableSource.java

```diff
@@ -0,0 +1,63 @@
+package org.apache.flink.mongodb.table.source;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.common.eventtime.WatermarkStrategy;
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.mongodb.table.sink.MongodbSinkConf;
+import org.apache.flink.table.catalog.ResolvedSchema;
+import org.apache.flink.table.connector.ChangelogMode;
+import org.apache.flink.table.connector.format.DecodingFormat;
+import org.apache.flink.table.connector.source.DynamicTableSource;
+import org.apache.flink.table.connector.source.ScanTableSource;
+import org.apache.flink.table.connector.source.SourceFunctionProvider;
+import org.apache.flink.table.data.RowData;
+
+@Internal
+public class MongodbDynamicTableSource implements ScanTableSource {
+    private final MongodbSinkConf mongodbSinkConf;
+    private final ResolvedSchema tableSchema;
```

Comment:
Please pass through the source and sink, whenever is possible, directly the `DataType` provided by `tableSchema.toSourceRowDataType()`, more than passing `ResolvedSchema` directly. In this case, you probably need `toPhysicalDataType` more than `toSourceDataType`, as suggested below by @Airblader.

Reply:
@slinkydeveloper I have modified the Table Sink according to what you said, so take a look and see if the logic I implemented meets your requirements.